doc/md/writing-motoko/randomness.md

```diff
@@ -0,0 +1,128 @@
+
+```motoko
+
+// Import the necessary modules, including the Random module:
+import Random = "mo:base/Random";
+import Char = "mo:base/Char";
+import Error = "mo:base/Error";
+
+// Define an actor
+
+actor {
+
+  // Define a stable variable that contains each card as a unicode character:
+  stable var deck : ?[var Char] = ?[var
+    '🂡','🂢','🂣','🂤','🂥','🂦','🂧','🂨','🂩','🂪','🂫','🂬','🂭','🂮',
+    '🂱','🂲','🂳','🂴','🂵','🂶','🂷','🂸','🂹','🂺','🂻','🂼','🂽','🂾',
+    '🃁','🃂','🃃','🃄','🃅','🃆','🃇','🃈','🃉','🃊','🃋','🃌','🃍','🃎',
+    '🃑','🃒','🃓','🃔','🃕','🃖','🃗','🃘','🃙','🃚','🃛','🃜','🃝','🃞',
+    '🃏'
+  ];
+
+  func bit(b : Bool) : Nat {
+    if (b) 1 else 0;
+  };
+
+  // Use a finite source of randomness defined as `f`.
+  // Return an optional random number between [0..`max`) using rejection sampling.
+  // A return value of `null` indicates that `f` is exhausted and should be replaced.
+  func chooseMax(f : Random.Finite, max : Nat) : ? Nat {
+    assert max > 0;
+    do ? {
+      var n = max - 1 : Nat;
+      var k = 0;
+      while (n != 0) {
+        k *= 2;
+        k += bit(f.coin()!);
+        n /= 2;
+      };
+      if (k < max) k else chooseMax(f, max)!;
+    };
+  };
+
+  // Define a function to shuffle the cards using `Random.Finite`.
+  public func shuffle() : async () {
+    let ?cards = deck else throw Error.reject("shuffle in progress");
+    deck := null;
+    var f = Random.Finite(await Random.blob());
+    var i : Nat = cards.size() - 1;
+    while (i > 0) {
+      switch (chooseMax(f, i + 1)) {
+        case (?j) {
+          let temp = cards[i];
+          cards[i] := cards[j];
+          cards[j] := temp;
+          i -= 1;
+        };
+        case null { // need more entropy
+          f := Random.Finite(await Random.blob());
+        }
+      }
+    };
+    deck := ?cards;
+  };
+
+  // Define a function to display the randomly shuffled cards.
+  public query func show() : async Text {
+    let ?cards = deck else throw Error.reject("shuffle in progress");
+    var t = "";
+    for (card in cards.vals()) {
+       t #= Char.toText(card);
+    };
+    return t;
+  }
+
+};
+```
+
+View this example on the [Motoko Playground](https://play.motoko.org/?tag=2675232834) or on [GitHub](https://github.com/crusso/card-shuffle/blob/main/src/cards_backend/main.mo).
+
+REMARK: The above solution directly uses the finite blob of 256-random bits returned by the management canister. Class `Random.Finite` uses this finite supply of bits to generate at most 256 coin flips, returning `null` when no more flips are possible.
+ 
+When its current supply of bits is exhausted, the code asynchronously requests another 256-bit blob to continue the shuffle. A more efficient, and equally robust approach would be to use the first 256-bit blob as a seed to a sequential pseudo random number generator, generating an infinite, lazy stream of bits, and then complete the shuffle with a single round of communication.
```

Comment:
```suggestion
:::tip

The above solution directly uses the finite blob of 256-random bits returned by the management canister. Class `Random.Finite` uses this finite supply of bits to generate at most 256 coin flips, returning `null` when no more flips are possible.
 
When its current supply of bits is exhausted, the code asynchronously requests another 256-bit blob to continue the shuffle. A more efficient, and equally robust approach would be to use the first 256-bit blob as a seed to a sequential pseudo random number generator, generating an infinite, lazy stream of bits, and then complete the shuffle with a single round of communication.

:::

```

Shall we change the remark to a tip so both paragraphs are in the same logical unit?